EventHubs migration guide (start)

richardpark-msft opened the pull request.

richardpark-msft committed:
starting the migration guide

chradek commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+| `EventHubClient.createFromConnectionString` | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [All](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/)
+| `EventHubClient.receive`                    | `EventHubConsumerClient.subscribe`                               | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.receiveBatch`               | Removed in favor of `EventHubConsumerClient.subscribe`           | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.createSender`               | `EventHubProducerClient.createProducer`                          | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
```
```suggestion
| `EventHubClient.createSender`               | `new EventHubProducerClient`                          | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
```
We require instantiating the producer client. Do you know which parameters changed from v2 to v5 as well? I think we've made some required, some optional, and renamed others (epoch is now ownerLevel for instance)

richardpark-msft replied:
Might be good to identify which options we think of as key - quite a few things shifted around so I'd really like to stick to documenting the core scenarios if possible.

As a part of this though - if we decide that these are important parameters we should probably have a sample demonstrating their usage regardless.

richardpark-msft committed:
Update event hubs migration guide.md

Co-Authored-By: chradek <51000525+chradek@users.noreply.github.com>

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+## Class changes
+
+`EventHubClient` has been split into two clients - `EventHubProducerClient` (for sending
+messages) and `EventHubConsumerClient` (for subscribing to messages).
```
`for subscribing to messages` -> `for receiving messages` to have a symmetry between send and receive?
I understand that the method name is `subscribe`, but while reading this line the customer is probably not yet familiar with the method names and might wonder what "subscribe" means?

richardpark-msft replied:
Makes sense. It makes so much sense that that was what I wrote in my first draft and then I was like "well, the new method is called subscribe" ;)

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+
+| In v2                                       | Equivalent in v5                                                 | Sample |
+|---------------------------------------------|------------------------------------------------------------------|--------|
+| `EventHubClient.createFromConnectionString` | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [All](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/)
```
What about when using AAD?

`EventHubClient.createFromAADTokenCredentials`  would become
`new EventHubProducerClient()` and `new EventHubConsumerClient()` using any of the [TokenCredentials](https://www.npmjs.com/package/@azure/identity)

richardpark-msft replied:
Awesome, will add.

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+|---------------------------------------------|------------------------------------------------------------------|--------|
+| `EventHubClient.createFromConnectionString` | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [All](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/)
+| `EventHubClient.receive`                    | `EventHubConsumerClient.subscribe`                               | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.receiveBatch`               | Removed in favor of `EventHubConsumerClient.subscribe`           | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
```
> Removed in favor of `EventHubConsumerClient.subscribe` which now supports batching. Default size is 10 which you can change using the option `maxBatchSize`

richardpark-msft replied:
The only reason I didn't add this text was because it crowds the cell a bit and the sample does clearly show how to use it (with batching).

Just trying to make the table work as an "at a glance" thing.

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+| `EventHubClient.createFromConnectionString` | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [All](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/)
+| `EventHubClient.receive`                    | `EventHubConsumerClient.subscribe`                               | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.receiveBatch`               | Removed in favor of `EventHubConsumerClient.subscribe`           | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.createSender`               | `new EventHubProducerClient`                          | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
```
I don't think `createSender()` existed in Track 1. Cant find it in the [reference docs](https://review.docs.microsoft.com/en-us/javascript/api/@azure/event-hubs/eventhubclient?view=azure-node-latest&branch=smoke-test-preview#methods)

richardpark-msft replied:
Nope, you're right. I'm not sure where I got that from.

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+| `EventHubClient.receive`                    | `EventHubConsumerClient.subscribe`                               | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.receiveBatch`               | Removed in favor of `EventHubConsumerClient.subscribe`           | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.createSender`               | `new EventHubProducerClient`                          | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
+| `EventHubClient.send`                       | `EventHubConsumerClient.sendBatch`                               | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
```
```suggestion
| `EventHubClient.send`                       | `EventHubConsumerClient.sendBatch`                               | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) which requires you to create a batch of events first.|
```

richardpark-msft replied:
You might be thinking of receiveEvents (not sendEvents.ts which does the sending).

There's a comment block in receiveEvents that walks you through running sendEvents to prime your EventHub.

ramya-rao-a replied:
My suggestion was more for the method `sendBatch` and not regarding the samples.

Current table contents tell the user that if they have been using `send`, they should now use `sendBatch`. But they also need to create a batch first, which they didnt need to do before. So, I believe its worthwhile calling that out.

ramya-rao-a commented on event hubs migration guide.md:
````diff
@@ -0,0 +1,81 @@
+
+For example, this code which receives from a partition in V2:
+
+```typescript
````
Add the line to create the client here as well to match with the v5 sample
```suggestion
```typescript
const client = EventHubClient.createFromConnectionString(connectionString);
```

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+const rcvHandler = client.receive(partitionId, onMessageHandler, onErrorHandler, {
+  eventPosition: EventPosition.fromStart(),
+  consumerGroup: "$Default"
+});
```
Add the line to stop the receiving of events to match with the v5 sample
```suggestion
});
await rcvHandler.stop();
```

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+| `EventHubClient.createSender`               | `new EventHubProducerClient`                          | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
+| `EventHubClient.send`                       | `EventHubConsumerClient.sendBatch`                               | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
+
+### Migrating code from `EventHubClient` to `EventHubConsumerClient`
```
```suggestion
### Migrating code from `EventHubClient` to `EventHubConsumerClient` for receiving events
```

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+See [`receiveEvents.ts`](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) 
+for a sample program demonstrating this.
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient`
```
```suggestion
### Migrating code from `EventHubClient` to `EventHubProducerClient` for sending events.
```

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,81 @@
+
+In V2, there were multiple options on how to send data.
+
+In V5, this has been consolidated into a more efficient `sendBatch` method.
```
Add few words as to why send batch is more efficient?

richardpark-msft replied:
Done.

ramya-rao-a commented on event hubs migration guide.md:
````diff
@@ -0,0 +1,81 @@
+
+So in V2:
+```typescript
+console.log(`Sending event: ${eventData.body}`);
````
Add the line to create the client here as well to match with the v5 sample

```suggestion
const client = EventHubClient.createFromConnectionString(connectionString);
console.log(`Sending event: ${eventData.body}`);
```

ramya-rao-a commented on event hubs migration guide.md:
````diff
@@ -0,0 +1,81 @@
+So in V2:
+```typescript
+console.log(`Sending event: ${eventData.body}`);
+await client.send(eventData, partitionId);
````
Since we want to highlight why sending with batches is better, should we consider the code snippet to reflect a function that takes an array of events? This is close to what we want to https://github.com/Azure/azure-sdk-for-js/issues/6254

The old sample would then be

```javascript
async function sendMyEvents(events) {
      await client.send(events); // Will fail if the total size of events exceed the max size supported by the library.
}
```

The newer one would be
```javascript
async function sendMyEvents(events) {
     const batch = await eventHubProducerClient.createBatch();
     let leftOverEvents = [];
     for (let i = 0; i < events.length; i++) {
        if (!batch.tryAdd(events[i]) {
               if (i === 0) {
                   // throw error here as we cannot batch the single event, too big
              }
               leftOverEvents = events.slice(i);
               break;
        }
     }
     await client.sendBatch(batch);
     await sendMyEvents(leftOverEvents);
}
```

ramya-rao-a reviewed (commented):
Great start!

I have left a few comments, please take a look

richardpark-msft committed:
Change wording to 'receiving'

richardpark-msft committed:
AAD line

richardpark-msft committed:
* Checked against code - casing for AAD was incorrect (should have been Aad).
* Remove createSender (doesn't exist!)

richardpark-msft committed:
Merge branch 'richardpark-eh-migration-guide' of https://github.com/richardpark-msft/azure-sdk-for-js into richardpark-eh-migration-guide

richardpark-msft committed:
Applying Ramya's changes

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

richardpark-msft committed:
Took part of a description from our REST documentation and tailored it a bit.

richardpark-msft committed:
Merge branch 'richardpark-eh-migration-guide' of https://github.com/richardpark-msft/azure-sdk-for-js into richardpark-eh-migration-guide

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,86 @@
+
+## Class changes
+
+`EventHubClient` has been split into two clients - `EventHubProducerClient` (for sending
```
We need to also talk about how the EPH class is not needed anymore and is replaced by EventHubConsumerClient

richardpark-msft replied:
Added a line (haven't done the sample yet)

richardpark-msft replied:
Looks like this is a duplicate comment to below. Resolving this one.

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,86 @@
+| `EventHubClient.receiveBatch`                  | Removed in favor of `EventHubConsumerClient.subscribe`           | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.send`                          | `EventHubConsumerClient.sendBatch`                               | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
+
+### Migrating code from `EventHubClient` to `EventHubConsumerClient` for receiving events
```
Need a section for 

> Migrating code from `EventProcessorHost` to `EventHubConsumerClient` for receiving events from all partitions

richardpark-msft replied:
Added - I think I captured the big differences:
* EPH used to have multiple error callbacks. Now there's one.
* EPH delivered a single message at a time. Now you get a batch of messages.
* Construction differences.

Let me know if there should be more - not as familiar with EPH.

richardpark-msft committed:
Updating migration guide sample for sending.

richardpark-msft committed:
Add in a description of EventProcessorHost

richardpark-msft committed:
Removed the 'as well'

mikeharder committed:
Unlink dependencies if tests fail (#6483)

- Results are published for both success and failure
- Unlink dependencies should use the same condition as publish
- Continuation of #6418

jeremymeng committed:
[Storage] Add more tests covering progress events (#6428)

* [Storage] Add more tests covering progress event

Related to #6352.

jsquire commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,118 @@
+# Migration Guide (EventHubs v2 to v5)
+
+## Class changes
```
It's not incorrect to call me verbose, but I think a short summary of why there is a new library and why I would want to migrate would help establish context and set expectations.

richardpark-msft replied:
I added some intro text - mostly just to make sure they understood if this was the document for them (EH v2 users that just need some quick snippets for migration) or if they're new to the library (then I point them to the main package readme.md)

jsquire commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,118 @@
+
+`EventProcessorHost`'s functionality has been merged into `EventHubConsumerClient`.
+
+| In v2                                          | Equivalent in v5                                                 | Sample |
```
This may be more helpful and align better with samples if it were framed in scenarios rather than focused on a client.  Especially in cases where there may be multiple samples that would be relevant and helpful.

richardpark-msft replied:
I broke them apart a bit just to give them a little separation - so I have receiving, sending and construction as three separate little tables instead of one single table.

jsquire commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,118 @@
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient` for sending events
+
+In V2, there were multiple options on how to send data.
```
It may be helpful for those without much experience to give some of the core options a passing mention.   _(unless the goal is to not draw attention for comparison)_

richardpark-msft replied:
@chradek had a similar point. 

My goal is to get them to the next stage of "oh, this is the call I need to use" without bogging down too much in each one. 

Do you have a list of ones you consider typical options?

jsquire replied:
I was thinking the most likely to have been used:

- Send a single event using defaults
- Send an iterable of events using defaults

richardpark-msft replied:
We have less variety than you in JS ;)

We only have the sendBatch() call at this point and it's covered by the inline sample we have under "Migrating code from `EventHubClient` to `EventHubProducerClient` for sending events"

So they're all a single case.

richardpark-msft committed:
[event-hubs] Better error handling and reporting (#6465)

Prior to this change we couldn't tell the difference between "storage is failing for some reason" and "we're just losing to other processors that are attempting to claim partitions".

Now an empty list of ownerships is okay and just indicates that we should do nothing this time around.

Fixes #6444 minus the the change to exit the processing loop.

jeremymeng committed:
[Core-http] Move getDefaultProxySettings into proxyPolicy (#6478)

* [Core-http] Move getDefaultProxySettings into proxyPolicy

so that libraries that don't use the PipelineOptions or createDefaultRequestPolicyFactories from core-http can use this behavior without duplicating code.

chradek committed:
[core-amqp] adds support for idle timeout and additional logging (#6492)

* [core-amqp] adds logging around dns.resolve in checkNetworkConnection

* [core-amqp] adds support for idle timeout check

* updates pnpm lock file

richardpark-msft committed:
[event-hubs] Updating samples folder to make it a standalone project (#6413)

Updating samples folder to make it a standalone project

* Makes it simpler for users to just download the entire folder, npm install and run.
* Makes it more effective for us as a testing tool since we can use it to test whatever is published.

richardpark-msft committed:
Adding example for EventProcessorHost => EventHubConsumerClient migration

richardpark-msft committed:
* Don't use $Default anywhere if we can avoid it. Guide the user choose a consumer group name.
* Give the document a bit more structure, including a TOC for the inline migration samples.

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,207 @@
+# Migration Guide (EventHubs v2 to v5)
+
+This document is intended for users that are familiar with EventHubs V2 (`@azure/event-hubs@2.x.x`) and wish 
```
```suggestion
This document is intended for users that are familiar with EventHubs V2 (`@azure/event-hubs@2.x.x` & `@azure/event-processor-host@2.x.x`) and wish 
```

richardpark-msft committed:
There were two packages, not one.

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,207 @@
+
+We've also merged the functionality from `EventProcessorHost` into 
+`EventHubConsumerClient`, allowing `EventHubConsumerClient` to be the single
+point of entry for receiving of any type within Event Hubs.
```
```suggestion
point of entry for receiving of any type (from single partition, all partitions, or with load balancing and checkpointing features) within Event Hubs.
```

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,207 @@
+point of entry for receiving of any type within Event Hubs.
+
+
+### Construction
```
```suggestion
### Client constructors
```

richardpark-msft committed:
Document the multiple types of message receiving we can handle.

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

richardpark-msft committed:
We only construct clients, not empires.

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,207 @@
+
+| In v2                                          | Equivalent in v5                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
+| `EventHubClient.createFromConnectionString`    | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts),  [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
```
```suggestion
| `EventHubClient.createFromConnectionString()`    | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts),  [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
```

richardpark-msft committed:
It's a function. Be proud of it and call it theoretically in the table.

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,207 @@
+| In v2                                          | Equivalent in v5                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
+| `EventHubClient.createFromConnectionString`    | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts),  [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
+| `EventHubClient.createFromAadTokenCredentials` | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [usingAadAuth](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/usingAadAuth.ts)
```
```suggestion
| `EventHubClient.createFromAadTokenCredentials()` | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [usingAadAuth](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/usingAadAuth.ts)
```

richardpark-msft committed:
Parens.

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,207 @@
+| `EventHubClient.createFromAadTokenCredentials` | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [usingAadAuth](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/usingAadAuth.ts)
+
+
+### Receiving events (subscribe)
```
I wouldnt suggest referring to "subscribe" in the header. If needed, you can add a one liner before the table about the fact that we now have a new name

```suggestion
### Receiving events 
```

richardpark-msft committed:
Removed!

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

Commit:
Updated with in-person feedback

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,206 @@
+];
+
+const client = EventHubClient.createFromConnectionString(connectionString);
+console.log(`Sending event: ${eventData.body}`);
```
`eventData` doesnt exist anymore

richardpark-msft replied:
Oops, no it does not. Also, was calling the wrong send API.

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,206 @@
+for (const event of eventsToSend) {
+  // messages can fail to be added to the batch if they exceed the maximum size configured for
+  // the EventHub.
+  const isAdded = batch.tryAdd({ body: event });
```
I would expect `eventsToSend` to already have the event data objects. So

```suggestion
  const isAdded = batch.tryAdd(event);
```

Commit:
Remove backtick from the toc links

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,206 @@
+      // if we can't add it and the batch is empty that means the message we're trying to send
+      // is too large, even when it would be the _only_ message in the batch.
+      //
+      // To fix this you'll need to split the message up across multiple batches or 
```
How would they split the message? Not sure if this is our recommendation though...
How about throwing an error?

richardpark-msft replied:
I changed the wording - splitting implies we can help them in some way but we really can't. I did mention they need to find a way to shrink the message or just skip it, hopefully making it clear that their future is in their hands.

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,206 @@
+
+    // and create a new one to house the next set of messages
+    batch = await producer.createBatch();
+    continue;
```
In this model, wont the last event that resulted in `isAdded` being `false` getting skipped?

richardpark-msft replied:
Only with a binary-based computer.

(fixed. Also, fixed the sample it was based on)

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,206 @@
+
+### Migrating code from `EventProcessorHost` to `EventHubConsumerClient` for receiving events
+
+In V2, `EventProcessorHost` allowed you to start receiving events, delivered via callbacks.
```
The selling point of EPH was the load balancing and checkpointing. We should call that out here

richardpark-msft replied:
This is less about what it's for (which they're familiar with) and more about how they achieve the same thing in the new library, right?

ramya-rao-a replied:
Yes, but if we are anyway talking about what EPH "allowed you to do", I'd prefer to have the full picture there if its doable. Something like

```suggestion
In V2, `EventProcessorHost` allowed you to balance the load between multiple instances of your program when receiving events.
```

ramya-rao-a replied:
Yes, but if we are anyway talking about what EPH "allowed you to do", I'd prefer to have the full picture there if its doable. Something like

```suggestion
In V2, `EventProcessorHost` allowed you to balance the load between multiple instances of your program when receiving events.
```

ramya-rao-a replied:
Yes, but if we are anyway talking about what EPH "allowed you to do", I'd prefer to have the full picture there if its doable. Something like

```suggestion
In V2, `EventProcessorHost` allowed you to balance the load between multiple instances of your program when receiving events.
```

Commit:
send example was incorrect (should have been sendBatch)

ramya-rao-a commented on event hubs migration guide.md:
```diff
@@ -0,0 +1,206 @@
+const subscription = eventHubConsumerClient.subscribe(
+  partitionId, {
+    // In V5 we deliver messages in batches, rather than a single message 
+    // at a time.
```
```suggestion
    // at a time. You can control the batch size via the options passed to the client.
```

richardpark-msft committed:
The user is in control of the batch size...

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

Commit:
Merge branch 'richardpark-eh-migration-guide' of https://github.com/richardpark-msft/azure-sdk-for-js into richardpark-eh-migration-guide

richardpark-msft committed:
Let's assume they're already full events and not just data.

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

Commit:
Tenatitively fixing sample

Commit:
Merge branch 'richardpark-eh-migration-guide' of https://github.com/richardpark-msft/azure-sdk-for-js into richardpark-eh-migration-guide

Commit:
Merge remote-tracking branch 'upstream/master' into richardpark-eh-migration-guide

richardpark-msft committed:
Advertise the goodness that was EPH and that is now part of the EventHubConsumerClient

Co-Authored-By: Ramya Rao <ramya.rao.a@outlook.com>

richardpark-msft committed:
* Make sample a bit more interesting by allowing them to specify the max batch size.
* Added in some more console output to better illustrate the control flow.
* Move migration guide to its final location.

ramya-rao-a commented:
The work here is being continued in https://github.com/Azure/azure-sdk-for-js/pull/6501
